Bento/Decorators/CustomInputComponent.swift

```diff
@@ -9,37 +10,66 @@ extension Renderable {
         return CustomInputComponent(
             source: self,
             customInput: input,
-            contentStatus: contentStatus,
-            highlightColor: highlightColor
+            focusEligibility: .eligible(contentStatus),
+            highlightColor: highlightColor,
+            focusesOnFirstDisplay: false
+        ).asAnyRenderable()
+    }
+
+    /// Sets up `customInput` to prepare for presention when user taps `self`,
+    /// and also presents it immediately when `state` is `.some`.
+    ///
+    /// - important: This method is useful when `customInput` needs to be displayed asynchronously after state change.
+    /// - note: Due to asynchronous presentation, `focusEligibility` is not supported.
+    public func customInput<State>(
+        immediatelyWhen state: State?,
```

Comment:
It could've been written as simply `input: CustomInput?`. Not sure what value the indirection adds here.

Reply:
There is already an existing `func customInput(_ input: CustomInput, ...)` , so removing this indirection from new method makes harder to distinguish the differences between 1. `customInput(input)`, 2. `customInput(.some(input))`, 3. `customInput(.none)`.

But I do agree that new method with `CustomInput?` argument is simpler and works like SwiftUI, so changing existing method is probably more preferred here, but I believe it's out of this PR's scope.

If you don't mind overlapping the similar method signature, I will change it.
What do you think?

Reply:
OK... `autodisplayingCustomInput(_:)`? I think the behaviour is different enough that giving it a different base name is probably better.

Reply:
Renamed in ef12cd7 .